openpype/plugins/publish/extract_review.py

```diff
@@ -1711,39 +1718,55 @@ def filter_output_defs(self, profile, subset_name, families):
 
         return filtered_outputs
 
-    def filter_outputs_by_tags(self, outputs, tags):
-        """Filter output definitions by entered representation tags.
+    def filter_outputs_by_custom_tags(self, outputs, custom_tags):
+        """Filter output definitions by entered representation custom_tags.
 
-        Output definitions without tags filter are marked as valid.
+        Output definitions without custom_tags filter are marked as invalid,
+        only in case representation is having any custom_tags defined.
 
         Args:
             outputs (list): Contain list of output definitions from presets.
-            tags (list): Tags of processed representation.
+            custom_tags (list): Custom Tags of processed representation.
 
         Returns:
             list: Containg all output definitions matching entered tags.
         """
         filtered_outputs = []
-        repre_tags_low = [tag.lower() for tag in tags]
+        repre_c_tags_low = [tag.lower() for tag in (custom_tags or [])]
         for output_def in outputs:
-            valid = True
-            output_filters = output_def.get("filter")
-            if output_filters:
-                # Check tag filters
-                tag_filters = output_filters.get("tags")
-                if tag_filters:
-                    tag_filters_low = [tag.lower() for tag in tag_filters]
-                    valid = False
-                    for tag in repre_tags_low:
-                        if tag in tag_filters_low:
-                            valid = True
-                            break
+            valid = False
+            tag_filters = output_def.get("filter", {}).get("custom_tags")
 
-                    if not valid:
-                        continue
+            if (
+                # if any of tag filter is empty, skip
+                custom_tags and not tag_filters
+                or not custom_tags and tag_filters
+            ):
+                continue
+            elif not custom_tags and not tag_filters:
+                valid = True
 
-            if valid:
-                filtered_outputs.append(output_def)
+            # lower all filter tags
+            tag_filters_low = [tag.lower() for tag in tag_filters]
+
+            self.log.debug("__ tag_filters: {}".format(tag_filters))
+            self.log.debug("__ repre_c_tags_low: {}".format(
+                repre_c_tags_low))
+
+            # check if any repre tag is not in filter tags
+            for tag in repre_c_tags_low:
+                if tag in tag_filters_low:
+                    valid = True
+                    break
+
+            if not valid:
+                continue
+
+            filtered_outputs.append(output_def)
```

Comment:
Removed unnecessary loggers and do tags filtering only if they are set.
```suggestion
            else:
                # lower all filter tags
                tag_filters_low = [tag.lower() for tag in tag_filters]

                # check if any repre tag is not in filter tags
                for tag in repre_c_tags_low:
                    if tag in tag_filters_low:
                        valid = True
                        break

            if valid:
                filtered_outputs.append(output_def)
```

Reply:
This will fix this issue too https://github.com/pypeclub/OpenPype/pull/4009#pullrequestreview-1150699781